### PR TITLE
test: Tab

### PR DESCRIPTION
**Pull request by yardz**
### SUMMARY
Tests for Tab component

### CODE REVIEW

File: superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  handleComponentDrop: jest.fn(),
+  updateComponents: jest.fn(),
+  setDirectPathToChild: jest.fn(),
+});
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+test('Render tab (no content)', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx:100]:
```suggestion
test('Render tab (no content)', () => {
```

File: superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  const props = creteProps();
+  props.renderType = 'RENDER_TAB';
+  render(<Tab {...props} />, { useRedux: true, useDnd: true });
+  expect(screen.getByText('🚀 Aspiring Developers')).toBeInTheDocument();
+  expect(EditableTitle).toBeCalledTimes(1);
+  expect(DragDroppable).toBeCalledTimes(1);
+});
+
+test('Render tab (no content) editMode:true', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx:109]:
```suggestion
test('Render tab (no content) editMode:true', () => {
```

File: superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  const props = creteProps();
+  props.editMode = true;
+  props.renderType = 'RENDER_TAB';
+  render(<Tab {...props} />, { useRedux: true, useDnd: true });
+  expect(screen.getByText('🚀 Aspiring Developers')).toBeInTheDocument();
+  expect(EditableTitle).toBeCalledTimes(1);
+  expect(DragDroppable).toBeCalledTimes(1);
+});
+
+test('Edit table title', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx:119]:
```suggestion
test('Edit table title', () => {
```

File: superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+  expect(EditableTitle).toBeCalledTimes(1);
+  expect(DragDroppable).toBeCalledTimes(1);
+
+  expect(props.updateComponents).not.toBeCalled();
+  userEvent.click(screen.getByText('🚀 Aspiring Developers'));
+  expect(props.updateComponents).toBeCalled();
+});
+
+test('Render tab (with content)', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx:133]:
```suggestion
test('Render tab (with content)', () => {
```

File: superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+      onResizeStart: expect.any(Function),
+      onResizeStop: expect.any(Function),
+      parentId: 'TAB-YT6eNksV-',
+    }),
+    {},
+  );
+  expect(DragDroppable).toBeCalledTimes(0);
+});
+
+test('Render tab (with content) editMode:true', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx:177]:
```suggestion
test('Render tab (with content) editMode:true', () => {
```

File: superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+      onResizeStart: expect.any(Function),
+      onResizeStop: expect.any(Function),
+      parentId: 'TAB-YT6eNksV-',
+    }),
+    {},
+  );
+  expect(DragDroppable).toBeCalledTimes(2);
+});
+
+test('Should call "handleDrop" and "handleTopDropTargetDrop"', async () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tab.test.tsx:222]:
```suggestion
test('Should call "handleDrop" and "handleTopDropTargetDrop"', () => {
```

**Review by michael-s-molina — CHANGES_REQUESTED** [review]

**Review by pkdotson — APPROVED** [review]
looks good to go thanks for the tests!

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**